### PR TITLE
Show periodic boundary conditions for masks and kernel

### LINKED ISSUE #728: `topology.PlotKernel` doesn't show periodic boundary conditions
**Description of problem:**
When plotting the kernel for a given neuron in a layer with periodic boundary conditions, the boundary conditions are not reflected in the plot.

**Version-Release or Git commit:**
 2.12.0

**How reproducible: (Always/Sometimes/Unsure)**
Always

**Steps to Reproduce:**
Run this script and view the plot `p`:

```python
import nest
from nest import topology as tp

layer_spec = {
    'extent': [2.0, 2.0],
    'rows': 10,
    'columns': 10,
    'elements': 'iaf_psc_alpha',
    'edge_wrap': True,
}

layer1 = tp.CreateLayer(layer_spec)
layer2 = tp.CreateLayer(layer_spec)

conn = {
    'connection_type': 'divergent',
    'mask': {'circular': {'radius': 0.75}},
    'kernel': {'gaussian': {'p_center': 1.0, 'sigma': 0.2}},
    'allow_autapses': False,
}

elt = tp.GetElement(layer1, [0, 0])

p = tp.PlotLayer(layer1)

tp.PlotKernel(p.axes[0], elt, conn['mask'])

p
```

**Actual results:**
kernel

**Expected results:**
A quadrant of the kernel should be shown at each corner, instead of just one.

### PR DESCRIPTION
**Pull request by stinebuu**
This PR makes `PlotKernel` plot periodic boundary conditions for masks and the gaussian kernel.

A few examples of how it will look like is given below, the source neuron is the one with a lighter red circle around itself. For each subplot in the figures, the source neuron is taken from a different part of the layer to make sure all boundary conditions are plotted. 

**Circular mask**
periodicboundarycircle
**Rotated elliptical mask**
periodicboundarytiltedellipse
**Gaussian Kernel**
periodicboundarykernel

This fixes #728.

### CODE REVIEW

File: topology/pynest/hl_api.py [hunk 1]
```diff
@@ -2099,23 +2110,53 @@ def PlotKernel(ax, src_nrn, mask, kern=None, mask_color='red',
     else:
         offs = np.array([0., 0.])
 
+    layer = GetLayer(src_nrn)
+    periodic = nest.GetStatus(layer)[0]['topology']['edge_wrap']
+    ext = nest.GetStatus(layer)[0]['topology']['extent']
```
Comment (heplesser) [inline topology/pynest/hl_api.py]: I would call this `extent`.

File: topology/pynest/hl_api.py [hunk 2]
```diff
@@ -1744,6 +1744,18 @@ def _draw_extent(ax, xctr, yctr, xext, yext):
            xticks=tuple(), yticks=tuple())
 
 
+def _get_newpos(pos, ext):
```
Comment (heplesser) [inline topology/pynest/hl_api.py]: I would call it `_shifted_positions()`, that also makes clear it returns multiple positions.

File: topology/pynest/hl_api.py [hunk 3]
```diff
@@ -2099,23 +2110,53 @@ def PlotKernel(ax, src_nrn, mask, kern=None, mask_color='red',
...
     if 'circular' in mask:
         r = mask['circular']['radius']
+
         ax.add_patch(plt.Circle(srcpos + offs, radius=r, zorder=-1000,
                                 fc='none', ec=mask_color, lw=3))
+
+        if periodic:
+            new_pos = _get_newpos(srcpos + offs, ext)
+
+            for pos in new_pos:
```
Comment (heplesser) [inline topology/pynest/hl_api.py]: I think here you can combine things a bit to
```
for pos in _shifted_positions(srcpos + offs, extent):
   ...
```

**Comment by wmayner** [comment]
Awesome, thank you!

**Pull request merged**